infrastructure/deployment/deploy.sh

```diff
@@ -269,6 +269,34 @@ split_and_join() {
    echo $SPLIT
 }
 
+pull_images_from_file() {
+    local file="$1"  # The file containing the Docker images
+    # Check if the file exists
+    if [[ ! -f "$file" ]]; then
+        echo "File not found: $file"
+        exit 1
+    fi
```

Comment:
If you generate code with ChatGPT (allowed, of course) please clean up formatting to match the rest of the file. Maybe there's a linter for shell scripts we could start using? The generated comments are usually pretty self-explanatory asa well so I tend to clean those away from the code just to reduce noise.

Besides that, it's good to properly read the generated code through. For instance in this code I highlighted, we are checking a file exists that we created a few steps backwards.